src/Adapter/Driver/Pdo/Pdo.php

```diff
@@ -298,7 +298,7 @@ public function getPrepareType()
     public function formatParameterName($name, $type = null)
     {
         if ($type === null && !is_numeric($name) || $type == self::PARAMETERIZATION_NAMED) {
-            return ':' . $name;
+            return ':' . preg_replace('/[^a-zA-Z0-9_]/', '_', $name);
```

Comment:
This replace can produce conflicts in bindParam name. For instance, if you have 2 fields `text-` and `text$` they will be replaced with `text_` and `text_` both.